client/state/global-sidebar/selectors.ts

```diff
@@ -1,11 +1,70 @@
 import { isPlansPageUntangled } from 'calypso/lib/plans/untangling-plans-experiment';
-import isScheduledUpdatesMultisiteRoute, {
+import {
 	isScheduledUpdatesMultisiteCreateRoute,
 	isScheduledUpdatesMultisiteEditRoute,
 } from 'calypso/state/selectors/is-scheduled-updates-multisite-route';
 import { isAdminInterfaceWPAdmin } from '../sites/selectors';
 import type { AppState } from 'calypso/types';
 
+export enum SidebarType {
+	None = 'none',
+	Global = 'global',
+	GlobalCollapsed = 'global-collapsed',
+	UnifiedSiteDefault = 'unified-site-default',
+	UnifiedSiteClassic = 'unified-site-classic',
+}
+
+interface GetSidebarTypeParams {
+	state: AppState;
+	siteId: number | null;
+	section: { group?: string };
+	route: string;
+}
+
+export function getSidebarType( params: GetSidebarTypeParams ): SidebarType {
+	const { state, siteId, section, route } = params;
+	const { group } = section;
+
+	if ( shouldShowSiteDashboard( params ) || shouldShowPluginDashboard( params ) ) {
+		return SidebarType.GlobalCollapsed;
+	}
+	if (
+		group === 'me' ||
+		group === 'reader' ||
+		group === 'sites-dashboard' ||
+		( group === 'sites' && ! siteId ) ||
```

Comment:
Should we account for scenarios where the user is not logged in?

Reply:
The code before the refactor didn't account for that 🤔 did you find any bug? I tested the live URL logged-out and everything seems fine 🙈

Reply:
Just feels weird that it returns the global sidebar even though it should be none for logged-out users 🙈